TNT-43002 Fix duplicate inline scripts added to DOM

XDex opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
When HTML offers with inline SCRIPT tags are rendered, these are added twice to page DOM, even though the scripts are actually executed just once on the page. We should fix it, so that no unexpected inline SCRIPT tags are added to page DOM after Personalization offers are applied.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
https://jira.corp.adobe.com/browse/TNT-43002

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
When HTML offers with inline SCRIPT tags are rendered, these are added twice to page DOM, even though the scripts are actually executed just once on the page. We should fix it, so that no unexpected inline SCRIPT tags are added to page DOM after Personalization offers are applied.

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

jonsnyder commented on src/components/Personalization/dom-actions/scripts.js:
```diff
@@ -80,8 +80,11 @@ export const getRemoteScriptsUrls = fragment => {
+export const executeInlineScripts = (parent, scripts) => {
+  scripts.forEach(script => {
+    parent.appendChild(script);
+    parent.removeChild(script);
```
How can you be sure the inline script is done executing? Do scripts that attach event handlers still work?

jonsnyder commented on test/unit/specs/components/Personalization/dom-actions/setHtml.spec.js:
```diff
@@ -49,13 +49,16 @@ describe("Personalization::actions::setHtml", () => {
+
+      const scriptElements = document.querySelectorAll("#evar123");
+      expect(scriptElements.length).toEqual(1);
     });
```
Can you add a test to see if an event handler defined in the inlineScript still works? And what if you have a setTimeout in the inline script?

jonsnyder reviewed (changes requested):
Can you add an integration test?

XDex commented:
> Can you add an integration test?

@jonsnyder I don't think an integration test is necessary, as we're only changing the rendering part here, which can be fully tested via unit tests, which I've added.

Note that I've also added some changes to unit test runner config, so async/await is now supported in unit tests.

jfkhoury commented:
@XDex Thank you did the changes to the unit tests and for making this change; a cleaner approach even if we weren’t double executing those script tags! 

I’m not sure how the unit test is verifying that the script is being removed. I might have missed it. 

One last thing, If @jonsnyder believes an integration test is helpful, let’s add one please; being extra safe never hurts, especially with those complicated Personalization features. No rush on this one since we can’t release anything anyway. 

Thanks again, it’s a pleasure to have you contribute to the project.

XDex commented:
@jfkhoury Thanks! I'll add a functional test as you suggest..
> I’m not sure how the unit test is verifying that the script is being removed. I might have missed it.

This is checked at line 62 in the test: `    expect(scriptElements.length).toEqual(1);`

ninaceban commented:
@XDex we have discussed about Alloy vs at.js in the past. Just because at.js does something in particular way, it doesn't mean that Alloy must follow suit.
I would argue that having scripts being part of the DOM both during `innerHTML` operation and after `appendChild` allows us to clearly see which DOM operation contained scripts and must be "treated" a little bit differently and it helps with debugging.

I would let @jfkhoury and @jonsnyder share their thoughts, but I think just documenting the behavior should be enough. We don't need to add more complexity to the existing DOM manipulation code.

XDex commented:
@ninaceban I have provided the motivation for this change in the linked [Jira ticket](https://jira.corp.adobe.com/browse/TNT-43002) comments, and I think @jfkhoury agrees that we should have this fixed.
@jfkhoury @jonsnyder Please take another look, I've also added an integration test.. Thank you!

jonsnyder reviewed (approved):
Thanks for adding the unit tests!

jfkhoury reviewed (approved):
Thank you for the tests.
